### PR TITLE
Seed firestore database

### PR DESCRIPTION
**Pull request by jbcden**
# What changed

This currently does not resolve any open issues but is helping work towards getting database seeding ready for firestore. In addition, this is adding some basic queries for the seeded data.

Currently there are some limitations with the current implementation. The current queries need to be updated to handle return the document id so that mutations can be applied (I think). In addition, the seeding is clumsy as it is difficult to associate resources together and running the seeding script would probably result in creating duplicate documents (there's probably a way around that but I'm not sure).

# Testing instructions

### CODE REVIEW

File: src/database.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  return healthcareProfessionals
+}
+
+export async function getFacilities() {
+    const db = getFirestore()
+    const facilitiesRef = db.collection('facilities')
+
+    const snapshot = await facilitiesRef.get()
+    var facilities = []
+    snapshot.forEach(doc => {
```
Comment (ermish) [inline src/database.ts:45]: Js optimization 
```suggestion
    const facilities = snapshot.map(doc => { ...}
```

File: src/database.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      facilities.push(doc.data())
+    })
+
+    facilities = Promise.all(facilities.map(facility => hydrateFacility(facility))).then((resource) => resource)
```
Comment (ermish) [inline src/database.ts:49]: Nice catch! This is the proper wait o await a foreach loop. BUT! They just released a new syntax to solve this ✨

```suggestion
    for await (const facility of facilities) {
    await yourcode ()
  }
```

File: src/database.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    // console.log(facilities)
+
+    return facilities
+}
+
+async function hydrateFacility(facility) {
+    const db = getFirestore()
+    const hpRef = db.collection('healthcareProfessionals')
+    const snapshot = await hpRef.where('id', 'in', facility.healthcareProfessionals).get()
```
Comment (ermish) [inline src/database.ts:58]: Nice! Looks like you figured out the queries ね

File: src/database.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    const snapshot = await hpRef.get()
+  var healthcareProfessionals = []
+    snapshot.forEach(doc => {
+      healthcareProfessionals.push(doc.data())
+    })
+
+  return healthcareProfessionals
+}
+
+export async function getFacilities() {
```
Comment (ermish) [inline src/database.ts:39]: For building queries/commands, we can move these to the respective domain services. `Facilityservice.ts`, etc. 

This allows the `database.ts` to have a single responsibility of managing db connections.

File: src/fakeData/facilities.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import { HealthcareProfessional, Locale, LocaleName, Degree, 
+    Specialty, SpecialtyName, SpokenLanguage, Insurance, Facility } from '../typeDefs/dbSchema'
+
+export const fakeFacilities = () => {
+    // const name : LocaleName = {
```
Comment (ermish) [inline src/fakeData/facilities.ts:5]: You could delete these comments since we have a different method to create a healthcare prof

**Review by ermish — APPROVED** [review]
Looks good so far! Feel free to merge and fix in follow up this time since we're not breaking anything if it's not running 😎

File: src/database.ts (hunk 1 above)
Comment (jbcden) [reply src/database.ts:45]: Surprisingly, while `snapshot.forEach` is defined here, `snapshot.map` is undefined.

**Pull request merged**